pkg/cluster/ipaddresses.go

```diff
@@ -255,10 +255,25 @@ func (m *manager) ensureGatewayCreate(ctx context.Context) error {
 			ImageRegistryStorageAccountName: m.doc.OpenShiftCluster.Properties.ImageRegistryStorageAccountName,
 		},
 	})
-	if err != nil && !cosmosdb.IsErrorStatusCode(err, http.StatusConflict) /* already exists */ {
+
+	recordExists := err != nil && cosmosdb.IsErrorStatusCode(err, http.StatusConflict)
+	if err != nil && !recordExists /* already exists */ {
 		return err
 	}
 
+	// ensure the record is this clusters if it exists
+	if recordExists {
```

Comment:
@bennerv When would this condition happen? If this is during cluster create, don't we just need to check that the record does not exist, and if it doesn't we create a new record

Reply:
@karanmagdani The gw record is created before this step and we need to ensure that it is for this specific cluster and it is not a stale/legacy record that is leftover.

https://github.com/Azure/ARO-RP/pull/1827/files/7b70040dd4e23084e4659ec6514ee083e8021842#diff-7369cd43feeb39bc03ab9c03efa6c5f85621d8554b94381e7d0a88398aca79a4R250

Reply:
Ah got it, makes sense! thank you @25region